### ISSUE TITLE
Adding support for std::array

### ISSUE THREAD

**Issue by wizmer** [BlueBrain/HighFive#100]
Hi,

It would be great if HighFive could support std::array as well :)

Thanks !

**Comment by joebarker87** [comment]
I've started coding this in a branch: https://github.com/joebarker87/HighFive/tree/feature/std_array

At the moment I've just covered basic writing of std::arrays and nested versions (i.e. std::array<std::array> and std::vector<std::array> etc.).

I didn't both with `#ifdef H5_USE_CXX11` guards because it looks like version 2.0 requires CXX11 anyway.

It's fairly straightforward to change the template specialisations and recursions. One potential problem will be that std::array cannot be resized. Upon reading the H5 file I think we will just have to throw an exception if the sizes don't match.

Any feedback or pointers from the main developers would be most welcome.
